tests/conftest.py

```diff
@@ -62,3 +63,24 @@
     finally:
         server.should_exit = True
         await task
+
+
+@pytest.fixture
+async def https_server():
+    here = os.path.dirname(__file__)
+    config = Config(
+        app=app,
+        lifespan="off",
+        ssl_keyfile=os.path.join(here, "assets/server.key"),
+        ssl_certfile=os.path.join(here, "assets/server.crt"),
+        port=8001,
+    )
+    server = Server(config=config)
+    task = asyncio.ensure_future(server.serve())
+    try:
+        while not server.started:
+            await asyncio.sleep(0.0001)
+        yield server
+    finally:
+        server.should_exit = True
+        await task
```

Comment:
Great! Want to submit *just* the https coverage stuff as a seperate pull request?
Also if you can document the steps you've used for generating the key/cert.